### PR TITLE
Added UI to remove posts as a moderator

### LINKED ISSUE #279: Moderator can remove posts
Remove is similar to delete, but slightly different. Only moderators can remove posts. Moderators can remove any posts, including their own. Moderators can also delete their own posts, hence moderators will see both "remove" and "delete" links on their own posts. On all other posts, moderators only have the remove option.

When a post is removed, it is no longer listed on the channel page. Users with the link can still go there if they paste the url in the browser—there, users will see a message that says "[Post removed]". They will still be able to read the comments on the page. 

Moderators will still be able to see the text of removed posts (although it will be grayed out). Removed posts can be un-removed by moderators by clicking "Approve". 

(See mocks below)

The API issue for this is #195.

image

image

**_After the post is removed, we will show a standard "Toast" message saying "Post Removed" (no mock for this.)_**

Post page as seen by moderator (comments will still be shown below this):
image

Post page as seen by user (comments will still be shown below this):
image

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #279 

#### What's this PR do?
Adds the UI for a moderator to remove posts

#### How should this be manually tested?
- As a regular user, verify you see no option to remove posts
- As a moderator verify you see the option to remove posts
- Remove a post, it should no longer show up in the listing
- As a moderator, you should be able to see the post content
- As a regular user you should see a message that the post has been removed
- As a moderator, you should be able to re-approve the post and it should return to a non-removed state  (shows up in channel listing and regular users can see the content)

### CODE REVIEW

**Comment by rhysyngsun** [comment]
I don't know what's up with Codecov today, but the coverage is pretty solid.

**Comment by noisecapella** [comment]
When I click the 'Remove' button I see a '#' appear in the URL... maybe there's a missing `preventDefault`?

**Comment by noisecapella** [comment]
I'm unable to see the post I removed in my own post list, though if I'm still on the page I can click to approve it. Not sure if we can do anything about it in this PR, I assume reddit has some special listing of removed posts for moderators

**Comment by rhysyngsun** [comment]
Yeah, posts are completely removed from listings for all users, nothing we can do short of modifying reddit upstream to fix that. Not sure on whether they're visible elsewhere.

File: static/js/components/ExpandedPostDisplay_test.js [hunk 1]
```diff
@@ -41,6 +43,8 @@ describe("ExpandedPostDisplay", () => {
     helper = new IntegrationTestHelper()
     post = makePost()
     beginEditingStub = sinon.stub()
```
Comment (noisecapella) [inline static/js/components/ExpandedPostDisplay_test.js]: Can you refactor this to use a sandbox instead?

File: static/js/components/material/Snackbar.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  snackbarRoot = null
+
+  props: SnackbarProps
+
+  componentDidMount() {
+    this.snackbar = new MDCSnackbar(this.snackbarRoot)
+  }
+
+  componentWillReceiveProps(nextProps: SnackbarProps) {
+    if (this.props.snackbar !== nextProps.snackbar) {
```
Comment (noisecapella) [inline static/js/components/material/Snackbar.js]: This will compare identity, not value. Is there something in ramda for deep equality?

**Comment by noisecapella** [comment]
I'm noticing if you click 'Remove' then go to another tab (at least in Firefox) and then go back, the snackbar stays up forever

Looks like this is in the MDC component: https://github.com/material-components/material-components-web/issues/1398

File: static/js/reducers/index_test.js [hunk 3]
```diff
@@ -256,6 +261,16 @@ describe("reducers", () => {
...
     })
+
+    it("should set snackbar message", async () => {
+      const payload = {
+        message: "hey there!"
+      }
+      const state = await dispatchThen(setSnackbarMessage(payload), [
+        SET_SNACKBAR_MESSAGE
+      ])
+      assert.equal(state.snackbar, payload)
```
Comment (noisecapella) [inline static/js/reducers/index_test.js]: This should be `deepEqual`

File: static/js/reducers/index_test.js [hunk 4]
```diff
@@ -256,6 +261,16 @@ describe("reducers", () => {
...
+    })
   })
 
   describe("subsribed channels (getChannels) reducer", () => {
```
Comment (noisecapella) [inline static/js/reducers/index_test.js]: Would you mind changing this to `subscribed` too?

File: static/js/components/material/Snackbar.js (hunk 2 above)
Comment (rhysyngsun) [reply static/js/components/material/Snackbar.js]: Ok, I'll need to add in something to guarantee object uniqueness per dispatch of `setSnackbarMessage` too.

Comment (rhysyngsun) [reply static/js/components/material/Snackbar.js]: I added a sequential id to the state, this should guarantee the above.

**Comment by rhysyngsun** [comment]
@noisecapella feedback addressed

**Pull request merged**